pkg/auth/auth.go

```diff
@@ -74,6 +88,21 @@ func systemContextWithOptions(sys *types.SystemContext, authFile, certDir string
 
 	if authFile != "" {
 		sys.AuthFilePath = authFile
+	} else {
+		// Keep this in sync with GetDefaultAuthFile()!
+		//
+		// Note that c/image does not natively implement the REGISTRY_AUTH_FILE
+		// variable, so not all callers look for credentials in this location.
+		if authFileVar := os.Getenv("REGISTRY_AUTH_FILE"); authFileVar != "" {
+			sys.AuthFilePath = authFileVar
+		}
+		// This preserves pre-existing _inconsistent_ behavior:
+		// If the Docker configuration exists in the default ~/.docker/config.json location,
+		// we DO NOT write to it; instead, we update auth.json in the default path.
+		// Only if the user explicitly sets DOCKER_CONFIG, we write to that config.json.
+		if dockerConfig := os.Getenv("DOCKER_CONFIG"); dockerConfig != "" {
```

Comment:
If REGISTRY_AUTH_FILE and DOCKER_CONFIG is set, then we use DOCKER_CONFIG?  We should probably at least logrus_debug this.

Reply:
Oops, I’ll fix that.

Reply:
Hopefully fixed.